core/src/main/java/io/hyperfoil/core/handlers/json/StreamQueue.java

```diff
@@ -5,118 +5,261 @@
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.jctools.util.Pow2;
 
 /**
  * This serves as an abstraction over several {@link ByteStream bytestreams} so that users can just append
  * new buffers to the back and use absolute positioning with single index.
  */
 public class StreamQueue {
-   protected static final Logger log = LogManager.getLogger(StreamQueue.class);
+   private static final Logger log = LogManager.getLogger(StreamQueue.class);
 
-   private final ByteStream[] parts;
+   private ByteStream[] parts;
    // indices used by users for the beginning (readerIndex) of the bytestream
-   private final int[] userIndex;
-   private final int mask;
+   private int[] userIndex;
+   private int mask;
    private int end = -1;
    private int tail = 0;
```

Comment:
I think that `tail` could be removed and just check the next(end) == `null` instead - I'll try to make this change on monday

Reply:
makes sense as the `tail` is always `end + 1`

Reply:
Yep, in the queues impl I made for JCtools we don't save both end and tail since the latter can be easily computed. In the lady commit version I made, it is like that already